### PR TITLE
Fix the "Show Tray Icon" preference

### PR DESCRIPTION
**Pull request by JBudny**
System: Windows 7
Fixed problem: File->Caprine Settings->Show Tray Icon option wasn't working.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you move the toggle logic into a function and also apply it here: https://github.com/sindresorhus/caprine/blob/a6c914ebde7390afb230de5f9daa8b55635fd3d4/source/menu.ts#L348 ?

**Comment by sindresorhus** [comment]
Also run `$ npm test` locally and ensure the linting passes.

File: source/config.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import Store = require('electron-store');
 import {is} from 'electron-util';
 
-type StoreType = {
+export type StoreType = {
```
Comment (sindresorhus) [inline source/config.ts]: Why are you exporting this? It's not used.

File: source/util.ts [hunk 2]
```diff
@@ -49,3 +50,34 @@ export function stripTrackingFromUrl(url: string): string {
 
 	return url;
 }
+
+export const toggleTrayIcon = (): void => {
+	config.set('showTrayIcon', !config.get('showTrayIcon'));
+	config.get('showTrayIcon') ? tray.create(getWindow()) : tray.destroy();
```
Comment (sindresorhus) [inline source/util.ts]: Save `config.get('showTrayIcon')` in a variable above and ten reuse it here. This should also use an if statement and not a ternary. Ternaries are not meant for function calls.

File: source/util.ts [hunk 3]
```diff
@@ -49,3 +50,34 @@ export function stripTrackingFromUrl(url: string): string {
...
+	if (config.get('launchMinimized')) {
+		if (!config.get('showTrayIcon')) {
+			toggleTrayIcon();
+		}
+
+		disableMenuItem(showTrayIconItem, true);
+
+		dialog.showMessageBox({
+			type: 'info',
+			message: 'Show Tray Icon option has been locked on enabled due to enabled Launch Minimized option.',
```
Comment (sindresorhus) [inline source/util.ts]:
```suggestion
			message: 'The “Show Tray Icon” setting is force-enabled while the “Launch Minimized” setting is enabled.',
```

File: source/config.ts (hunk 1 above)
Comment (JBudny) [reply source/config.ts]: I've cherry-picked this from another branch I closed and improved it a little. Somehow I forget to delete this export. Thanks for catching it.

File: source/util.ts (hunk 2 above)
Comment (JBudny) [reply source/util.ts]: Thank you for pointing me out that misuse of ternary operators. I wasn't aware of it. And sorry you had to wait so long for those changes.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**